app/components/login/Login.js

```diff
@@ -1,42 +1,34 @@
 'use strict';
 import React, {Component, PropTypes} from 'react';
 
-import {View, Text, Linking} from 'react-native';
+import {View, Text, Linking, WebView} from 'react-native';
 import {Actions} from 'react-native-router-flux';
 
 export default class Login extends Component {
 
-    componentDidMount() {
-
-        let stravaUrl = 'https://www.strava.com/oauth/authorize?client_id=15688&state=mystate&response_type=code&redirect_uri=strava://localhost&scope=public&approval_prompt=auto';
-
-        Linking.addEventListener('url', this.handleOpenURL);
-
-        Linking.canOpenURL(stravaUrl).then(supported => {
-            if (supported) {
-                Linking.openURL(stravaUrl);
-            } else {
-                console.log('Don\'t know how to open URI: ' + this.props.url);
+    handleNavigationChange(event){
+        if(this.refs != null){
+            if (event.url.indexOf('strava://localhost') === -1) {

```

Comment:
use a ref callBack https://facebook.github.io/react/docs/refs-and-the-dom.html#the-ref-callback-attribute